pkg/expr/sql/parser.go

```diff
@@ -1,89 +1,68 @@
 package sql
 
 import (
-	"encoding/json"
 	"fmt"
 	"sort"
-	"strings"
 
+	"github.com/dolthub/vitess/go/vt/sqlparser"
 	"github.com/grafana/grafana/pkg/infra/log"
-	"github.com/jeremywohl/flatten"
-)
-
-const (
-	TABLE_NAME    = "table_name"
-	ERROR         = ".error"
-	ERROR_MESSAGE = ".error_message"
 )
 
 var logger = log.New("sql_expr")
 
 // TablesList returns a list of tables for the sql statement
 func TablesList(rawSQL string) ([]string, error) {
-	db := NewInMemoryDB()
-	rawSQL = strings.Replace(rawSQL, "'", "''", -1)
-	cmd := fmt.Sprintf("SELECT json_serialize_sql('%s')", rawSQL)
-	ret, err := db.RunCommands([]string{cmd})
-	if err != nil {
-		logger.Error("error serializing sql", "error", err.Error(), "sql", rawSQL, "cmd", cmd)
-		return nil, fmt.Errorf("error serializing sql: %s", err.Error())
-	}
-
-	ast := []map[string]any{}
-	err = json.Unmarshal([]byte(ret), &ast)
+	stmt, err := sqlparser.Parse(rawSQL)
 	if err != nil {
-		logger.Error("error converting json sql to ast", "error", err.Error(), "ret", ret)
-		return nil, fmt.Errorf("error converting json to ast: %s", err.Error())
+		logger.Error("error parsing sql: %s", err.Error(), "sql", rawSQL)
+		return nil, fmt.Errorf("error parsing sql: %s", err.Error())
 	}
 
-	return tablesFromAST(ast)
-}
+	tables := make(map[string]struct{})
 
-// tablesFromAST returns a list of tables from the ast
-func tablesFromAST(ast []map[string]any) ([]string, error) {
-	flat, err := flatten.Flatten(ast[0], "", flatten.DotStyle)
-	if err != nil {
-		logger.Error("error flattening ast", "error", err.Error(), "ast", ast)
-		return nil, fmt.Errorf("error flattening ast: %s", err.Error())
-	}
-
-	tables := []string{}
-	for k, v := range flat {
-		if strings.HasSuffix(k, ERROR) {
-			v, ok := v.(bool)
-			if ok && v {
-				logger.Error("error in sql", "error", k)
-				return nil, astError(k, flat)
-			}
-		}
-		if strings.Contains(k, TABLE_NAME) {
-			table, ok := v.(string)
-			if ok && !existsInList(table, tables) {
-				tables = append(tables, v.(string))
+	walkSubtree := func(node sqlparser.SQLNode) error {
+		err = sqlparser.Walk(func(node sqlparser.SQLNode) (kontinue bool, err error) {
+			switch v := node.(type) {
+			case *sqlparser.AliasedTableExpr:
+				if tableName, ok := v.Expr.(sqlparser.TableName); ok {
+					tables[tableName.Name.String()] = struct{}{}
+				}
+			case *sqlparser.TableName:
+				tables[v.Name.String()] = struct{}{}
+			case *sqlparser.FuncExpr:
+				switch v.Name.Lowered() {
+				case "load_file", "sleep", "connection_id", "user":
```

Comment:
i think you could implement [FunctionProvider](https://github.com/dolthub/go-mysql-server/blob/67aa2a4303668833b365b7a67da56765379b7164/sql/functions.go#L27-L32) and filter the [built in functions](https://github.com/dolthub/go-mysql-server/blob/67aa2a4303668833b365b7a67da56765379b7164/sql/expression/function/registry.go#L34)

Reply:
Ya I have that in my notes to follow up with, tried that the other week and somehow got stuck and haven't revisted yet